File: apps/ios/Models/Room.swift
import Foundation

enum RoomStatus: String, Codable {
    case waiting
    case active
    case closed
}

struct RoomSettings: Codable {
    var sourceLanguage: String
    var targetLanguage: String
    var romajiEnabled: Bool
    var suggestionsEnabled: Bool
    var maxParticipants: Int

    static let defaults = RoomSettings(
        sourceLanguage: "ja",
        targetLanguage: "en",
        romajiEnabled: true,
        suggestionsEnabled: true,
        maxParticipants: 10
    )
}

struct Room: Identifiable, Codable {
    let id: String
    let joinCode: String
    var status: RoomStatus
    var settings: RoomSettings
    let hostId: String
    let createdAt: Date
    var closedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case joinCode, status, settings, hostId, createdAt, closedAt
    }
}

File: packages/shared-types/src/room.ts
export type RoomStatus = "waiting" | "active" | "closed";

export interface RoomSettings {
  /** Source language for the room (e.g. "ja") */
  sourceLanguage: string;
  /** Target language for translation (e.g. "en") */
  targetLanguage: string;
  /** Whether romaji generation is enabled */
  romajiEnabled: boolean;
  /** Whether suggestion generation is enabled */
  suggestionsEnabled: boolean;
  /** Maximum number of participants allowed */
  maxParticipants: number;
}

export interface Room {
  _id: string;
  /** Short code participants use to join */
  joinCode: string;
  status: RoomStatus;
  settings: RoomSettings;
  /** ID of the host participant */
  hostId: string;
  createdAt: number;
  closedAt?: number;
}
